tests/suites/os/tests/purge-data/index.js

```diff
@@ -87,90 +69,86 @@ module.exports = {
 				const testFile = '/mnt/data/resin-data/reset-check';
 				const resetFile = '/mnt/data/remove_me_to_reset';
 
-				test.comment(`Waiting for engine service to be active...`);
-				await this.utils.waitUntil(async () => {
-					return (
-						(await this.context
-							.get()
-							.worker.executeCommandInHostOS(
-								`systemctl is-active balena.service`,
-								this.link,
-							)) === 'active'
-					);
-				}, false, 2 * 60 * 4, 250);
-
-				test.comment(`Waiting for supervisor service to be active...`);
-				await this.utils.waitUntil(async () => {
-					return (
-						(await this.context
-							.get()
-							.worker.executeCommandInHostOS(
-								`systemctl is-active balena-supervisor.service`,
-								this.link,
-							)) === 'active'
-					);
-				}, false, 2 * 60 * 4, 250);
+				await test.resolves(
+					this.systemd.waitForServiceState(
+						'balena-supervisor.service',
+						'active',
+						this.link
+					),
+					'Should wait for balena.service to be active'
+				);
+
+				await test.resolves(
+					this.systemd.waitForServiceState(
+						'balena-supervisor.service',
+						'active',
+						this.link
+					),
+					'Should wait for balena-supervisor.service to be active'
+				);
```

Comment:
```suggestion
				await test.resolves(
				  Promise.all([
				    this.systemd.waitForServiceState('balena-supervisor.service','active',this.link), 
						this.systemd.waitForServiceState('balena.service','active',this.link)
				  ]),
				  'Should wait for balena.service or balena-supervisor.service to be active'
				 )
				
```
Should be cleaner, faster with this having 2 promises running in parallel - https://javascript.info/promise-api#promise-all 
Also, I think you copied the same service name twice. Fixed that to have engine and SV services getting checked. Can you give the alignment a fix too. 
Do give this a test, I am writing this from memory and using a guide directly into GithUb.

The drawback for this is we won't know which one is the culprit when they are not getting active. I feel a little look into Promise.all can help us get better errors.